pycloudlib/azure/security_types.py

```diff
@@ -0,0 +1,136 @@
+"""Azure Security Types Classes."""
+import abc
+import re
+from enum import Enum
+from typing import Any, Dict
+
+from pycloudlib import util
+
+
+class AzureCVMOSDiskEncryption(Enum):
+    """Represents Azure OS disk encryption types."""
+
+    VM_GUEST_STATE_ONLY = "VMGuestStateOnly"
+    DISK_WITH_VM_GUEST_STATE = "DiskWithVMGuestState"
+
+
+class SecurityType(abc.ABC):
+    """Base Azure Security Type Class."""
+
+    @abc.abstractmethod
+    def configure_vm_params(
+        self, vm_params: Dict[str, Any], **kwargs
+    ) -> Dict[str, Any]:
+        """Configure the vm_params to be passed to Azure.
+
+        Args:
+            vm_params: dict, The parameters passed to Azure
+            **kwargs: dictionary of security type parameters
+        """
+        raise NotImplementedError
+
+    def is_correct_vm_size(self, vm_size: str) -> bool:
+        """Check if the vm size is supported by security type.
+
+        Args:
+            vm_size: string, The
```

Comment:
What is the format / units of `vm_size` ?

Reply:
The `vm_size` is a string

Reply:
What are the valid inputs for vm_size? Its a string, but a VERY specific string that has different meanings. E.g., it could actually be a `CloudImageSize` which is a string that must be formatted as `integerSizeIdentifer` (e.g. 8gb).

Reply:
After discussion on standup, it is better not to include this check and let the user provide a valid vm_size. Furthermore, Azure has a good error message incase of an unsupported vm size.